app/models/base_channel_details.rb

```diff
@@ -0,0 +1,17 @@
+class BaseChannelDetails < ActiveRecord::Base
+  self.abstract_class = true
+
+  has_one :channel, as: :details
+
+  after_initialize :initialize_id, unless: -> { self.id }
+
+  def channel_identifier
+    raise "Override channel_identifier"
+  end
+
+  private
+
+  def initialize_id
```

Comment:
Would you explain what this is for?

Reply:
This was left over from Larry's initial PR.  I couldn't figure out what he wanted it for, but forgot to remove it.  Will delete.